lib/file.go

```diff
@@ -122,6 +127,39 @@ func ReadDatasetFile(path string) (ds *dataset.Dataset, err error) {
 	return
 }
 
+func fillDatasetOrComponent(fields map[string]interface{}, path string, ds *dataset.Dataset) error {
+	switch fields["qri"] {
```

Comment:
This really needs to be documented better, but switching on the qri field ("component kinds"), should be examining only the first two characters of kind: https://github.com/qri-io/dataset/blob/f2ced13e41b51fe4f2549d8ed6d585a871659333/kind.go#L13

First, just to formalize the spec, the `qri` field, if present MUST have a value in the form:
```
component_acronym ':' version_number
```
where `component_acronym` MUST be a unique two character identifier for the type of component, and `version_number` is an unsigned integer.

The version number might change, and maintaining code is irritating. So how about:

```go
func fillDatasetOrComponent(fields map[string]interface{}, path string, ds *dataset.Dataset) (err error) {
	var fill interface{}
	fill = ds

	if kindStr, ok := fields["qri"].(string); ok && len(kindStr) > 3 {
		switch kindStr[:2] {
		case "md":
			ds.Meta = &dataset.Meta{}
			fill = ds.Meta
		case "cm":
			ds.Commit = &dataset.Commit{}
			fill = ds.Commit
		case "st":
			ds.Structure = &dataset.Structure{}
			fill = ds.Structure
		}
	}

	if err = base.FillStruct(fields, fill); err != nil {
		return err
	}
	absDatasetPaths(path, ds)
	return nil
}
```

I'm not a massive fan of the `absDatasetPaths(path,ds)` at the end of this func, but after reading, it's clear this is the best place for it currently, so let's keep it.